code/controllers/ShareDraftController.php

```diff
@@ -42,25 +42,50 @@ public function preview(SS_HTTPRequest $request) {
 			return $this->errorPage();
 		}
 
-		$page = $shareToken->Page();
```

Comment:
In some cases this was returning a `$page` with ID = 0, so change this to check for stage version explicitly.